apps/full-site-editing/full-site-editing-plugin/editor-gutenboarding-launch/index.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import domReady from '@wordpress/dom-ready';
+import { addAction } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import './index.scss';
+
+interface CalypsoifyWindow extends Window {
+	calypsoifyGutenberg?: {
+		frankenflowUrl?: string;
+		isFseGutenboarding?: boolean;
+		[ key: string ]: unknown;
+	};
+}
+declare const window: CalypsoifyWindow;
+
+domReady( () => {
+	updateEditor();
+	// Hook fallback incase setGutenboardingStatus runs after initial dom render.
+	addAction( 'setGutenboardingStatus', 'a8c-gutenboarding', updateEditor );
```

Comment:
Maybe move the condition from `updateEditor`? 

```suggestion
if (
  !window?.calypsoifyGutenberg?.isFseGutenboarding ||
  !window?.calypsoifyGutenberg?.frankenflowUrl
) {
  addAction("setGutenboardingStatus", "a8c-gutenboarding", updateEditor);
} else {
  updateEditor();
}

```
I'm assuming this condition means the hook isn't needed.

This way you won't add the action unless needed.

Reply:
The values are potentially set between domReady and when the action is fired. The goal is to:

- Run as early as possible
- Run 0 or 1 times

We can move the conditional checks around, but it's important to check them both places.

Reply:
Then I think as is is better.